lnwallet/channel.go

```diff
@@ -3366,21 +3377,60 @@ func (lc *LightningChannel) createCommitDiff(newCommit *commitment,
 	// disk.
 	diskCommit := newCommit.toDiskCommit(lntypes.Remote)
 
+	// We need to limit the size of the custom records to prevent the whole
+	// commitment_signed message to not exceed the maximum message size of
+	// 65k. If we assume a maximum number of HTLCs on the commitment of 483,
+	// and each signature being 64 bytes, we already use up 30k of the
```

Comment:
Ah, right... I'm not sure we can split the sigs over multiple messages. Or it would be a big refactor.

I changed the code to calculate the remaining number of bytes left for custom records. If we exceed that, we'll just fail the HTLC (I think). So for custom channels the real number of potential in-flight HTLCs will be quite a bit lower than the theoretical total of 966. But I think that's fine for now.
What do you think?

cc @Roasbeef

Reply:
I don't know if we can calculate the space dynamically here. We have to avoid a situation where `createCommitDiff` fails as it can cause the link to fail with `ErrInternalError` which causes us to send an `Error` to our peer and make them force close on us. This is made worse if a 3rd party can route HTLCs through us, causing both sides to hit the `max_accepted_htlcs` limit and then potentially triggering channel failure due to the custom records size.